src/modules/70-pipeline/utils/templateUtils.tsx

```diff
@@ -98,6 +135,15 @@ export const createTemplate = <T extends PipelineInfoConfig | StageElementConfig
       if (template.versionLabel) {
         set(draft, 'template.versionLabel', template.versionLabel)
       }
+      if (template.gitDetails?.branch && template.gitDetails?.repoName) {
+        if (
```

Comment:
This if conditions can be move to the upper if with &&. 
```suggestion
        if ((template.gitDetails?.branch && template.gitDetails?.repoName) && ((template.gitDetails.repoName === repoName && template.gitDetails.branch !== branch) ||
          (template.gitDetails.repoName !== repoName &&
            template.gitDetails.branch !== template.gitDetails?.defaultBranch)))
```